api/v1beta1/database_webhook.go

```diff
@@ -107,6 +119,37 @@ func ValidateSecretTemplates(templates map[string]string) error {
 	return nil
 }
 
+func ValidateTemplates(templates Templates) error {
+	for _, template := range templates {
+		fmt.Printf("%s - %s\n", template.Name, template.Template)
+		helpers := []string{"Protocol", "Host", "Port", "Password", "Username", "Password", "Database"}
+		functions := []string{"Secret", "ConfigMap", "Query"}
+		// This regexp is getting fields from mustache templates so then they can be compared to allowed fields
+		reg := "{{\\s*\\.([\\w\\.]+)\\s*(.*?)\\s*}}"
+		fmt.Println(reg)
+		r, _ := regexp.Compile(reg)
+		fields := r.FindAllStringSubmatch(template.Template, -1)
+		for _, field := range fields {
+			fmt.Printf("1-%s\n2-%s\n", field[1], field[2])
+			if slices.Contains(helpers, field[1]){
```

Comment:
hard to read the code. think maybe wrap this in more understandable functions?

```go
if validHelperFunction(field[1]){
  continue
}
```

Reply:
It should look better now